internal/test/smoke/suite_base_test.go

```diff
@@ -37,6 +37,30 @@ import (
 	client_rbac "github.com/podengo-project/idmsvc-backend/internal/usecase/client/rbac"
 )
 
+type XRHIDProfile string
+
+const (
+	// XRHIDNothing does not make any action, letting a progressive change
+	XRHIDNothing XRHIDProfile = ""
+	// XRHIDNone no XRHID header is added
+	XRHIDNone XRHIDProfile = "None"
+	// XRHIDUser add a user XRHID header
+	XRHIDUser XRHIDProfile = "User"
+	// XRHIDServiceAccount add a service account XRHID header
+	XRHIDServiceAccount XRHIDProfile = "ServiceAccount"
+	// XRHIDSystem add a system XRHID header
+	XRHIDSystem XRHIDProfile = "System"
```

Comment:
:+1: I have re-written in a different way; and you are right, basically it is an enumeration of supported xrhid profiles; **but** two values have a special meaning; the empty string fit when we don't initialize the value in a structure, the "None" has another special meaning that assure to delete the xrhid. They are chosen on purpose, to avoid the empty string because a forgotten and evoke to delete the xrhid header without that intention, and to delete specifically with the second value when required.